helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/WagedInstanceCapacity.java

```diff
@@ -130,28 +136,29 @@ public void processPendingMessages(ResourceControllerDataProvider cache,
     }
   }
 
-
-  private void processCurrentState(ResourceControllerDataProvider cache,
-      CurrentStateOutput currentStateOutput, Map<String, Resource> resourceMap,
-      WagedResourceWeightsProvider weightProvider) {
+  void processCurrentState(ResourceControllerDataProvider cache,
+      CurrentStateOutput currentStateOutput, Map<String, Resource> resourceMap) {
 
     // Iterate through all the resources
     for (Map.Entry<String, Resource> entry : resourceMap.entrySet()) {
       String resName = entry.getKey();
       Resource resource = entry.getValue();
 
-      // if Resource is WAGED managed, then we need to manage the capacity.
-      if (!WagedValidationUtil.isWagedEnabled(cache.getIdealState(resName))) {
-        continue;
-      }
-
       // list of partitions in the resource
       Collection<Partition> partitions = resource.getPartitions();
 
+      // calculate merge resource-config one for each resource.
+      ResourceConfig resourceConfig = cache.getResourceConfig(resName);
+      IdealState is = cache.getIdealState(resName);
+      ResourceConfig mergedResourceConfig =
+          ResourceConfig.mergeIdealStateWithResourceConfig(resourceConfig, is);
+
       for (Partition partition : partitions) {
         String partitionName = partition.getPartitionName();
         // Get Partition Weight
-        Map<String, Integer> partCapacity = weightProvider.getPartitionWeights(resName, partitionName);
+        Map<String, Integer> partCapacity =
+            WagedRebalanceUtil.fetchCapacityUsage(partitionName, mergedResourceConfig, cache.getClusterConfig());
```

Comment:
i have 2 questions: 1. do we need to update mergedResourceConfig everytime?   2.  resource/partition weight doesn't change frequently. so where are we caching this value?

Reply:
So earlier, we were computing `mergedResourceConfig` everytime i.e. for each partition of a resource. Now, we are computing this at resource level, and using same for all partitions.

We can cache it as well. Can I do it as part of next PR as to keep the size of this limited and bringing one change at a time..

Reply:
you can check it in next change. Also based on Junkai comment, please evaulate if we even need it?

Reply:
Yes, I have removed this.